src/components/frame/v5/pages/StreamingPaymentsPage/partials/StreamingPaymentsTable/StreamingPaymentsTable.tsx

```diff
@@ -0,0 +1,126 @@
+import { BigNumber } from 'ethers';
+import React from 'react';
+import { defineMessages } from 'react-intl';
+
+import { DEFAULT_TOKEN_DECIMALS } from '~constants';
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import { useSearchStreamingPaymentsQuery } from '~gql';
+import { notNull } from '~utils/arrays/index.ts';
+import { formatText } from '~utils/intl.ts';
+import { convertToMonthlyAmount } from '~utils/streamingPayments.ts';
+import Table from '~v5/common/Table/Table.tsx';
+
+import { useRenderSubComponent, useStreamingTableColumns } from './hooks.tsx';
+import { type StreamingTableFieldModel } from './types.ts';
+
+const displayName =
+  'pages.StreamingPaymentsPage.partials.StreamingPaymentsTable';
+
+const MSG = defineMessages({
+  title: {
+    id: `${displayName}.title`,
+    defaultMessage: 'Streaming Payments',
+  },
+});
+
+const StreamingPaymentsTable = () => {
+  const {
+    colony: { nativeToken, colonyAddress },
+  } = useColonyContext();
+  const { data, loading } = useSearchStreamingPaymentsQuery({
+    variables: {
+      colonyId: colonyAddress,
+    },
+  });
+
+  const groupedStreamingPayments: StreamingTableFieldModel[] = (
+    data?.searchStreamingPayments?.items || []
+  )
+    .filter(notNull)
+    .map((item) => ({
+      title: item.actions?.items[0]?.metadata?.customTitle || '',
+      token: item.token || nativeToken,
+      nativeDomainId: item.nativeDomainId,
+      paymentId: item.id,
+      amount: item.amount,
+      creatorAddress: item.creatorAddress,
+      recipientAddress: item.recipientAddress,
+      interval: item.interval,
+      transactionId: item.actions?.items[0]?.transactionHash || '',
+    }))
```

Comment:
I was actually hoping you could spread the original streaming payment object and just add the fields you're missing/need easier access to. That way we avoid fallbacks at the data layer, and can provide them right where they are rendered in the UI if needed.

Reply:
@jakubcolony This is being addressed in - https://github.com/JoinColony/colonyCDapp/pull/3379